flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceScanFetcher.java

```diff
@@ -113,6 +113,7 @@ public Iterator<SourceRecord> pollSplitRecords() throws InterruptedException {
             boolean reachBinlogEnd = false;
             final List<SourceRecord> sourceRecords = new ArrayList<>();
             while (!reachBinlogEnd) {
+                checkReadException();
```

Comment:
I think it solves the problem #1098. It is better that add a test for JdbcSourceScanFetcher.java like com.ververica.cdc.connectors.mysql.debezium.reader.SnapshotSplitReaderTest#testThrowRuntimeExceptionInSnapshotScan